src/utils/isSubsetOf.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { ReadonlySetLike, isSubsetOf } from "./isSubsetOf";
+
+// Tests are adapted from:
+// https://github.com/zloirock/core-js/blob/227a758ef96fa585a66cc1e89741e7d0bb696f48/tests/unit-global/es.set.is-subset-of.js
+
+describe("isSubsetOf", () => {
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    let s1: Set<any>;
+    let s2: ReadonlySetLike<unknown>;
+
+    it("should implement isSubsetOf correctly", () => {
+        s1 = new Set([1]);
+        s2 = new Set([1, 2, 3]);
+        expect(isSubsetOf(s1, s2)).toBe(true);
+
+        s1 = new Set([1]);
+        s2 = new Set([2, 3, 4]);
+        expect(isSubsetOf(s1, s2)).toBe(false);
+
+        s1 = new Set([1, 2, 3]);
+        s2 = new Set([5, 4, 3, 2, 1]);
+        expect(isSubsetOf(s1, s2)).toBe(true);
+
+        s1 = new Set([1, 2, 3]);
+        s2 = new Set([5, 4, 3, 2]);
+        expect(isSubsetOf(s1, s2)).toBe(false);
+
+        s1 = new Set([1]);
+        s2 = createSetLike([1, 2, 3]);
+        expect(isSubsetOf(s1, s2)).toBe(true);
+
+        s1 = new Set([1]);
+        s2 = createSetLike([2, 3, 4]);
+        expect(isSubsetOf(s1, s2)).toBe(false);
+
+        s1 = new Set([1, 2, 3]);
+        s2 = createSetLike([5, 4, 3, 2, 1]);
+        expect(isSubsetOf(s1, s2)).toBe(true);
+
+        s1 = new Set([1, 2, 3]);
+        s2 = createSetLike([5, 4, 3, 2]);
+        expect(isSubsetOf(s1, s2)).toBe(false);
+
+        s1 = new Set([1, 2, 3]);
+        s2 = new Set([1]);
+        expect(isSubsetOf(s1, s2)).toBe(false);
+
+        s1 = new Set([1, 2, 3]);
+        s2 = new Set();
+        expect(isSubsetOf(s1, s2)).toBe(false);
+
+        s1 = new Set();
+        s2 = new Set([1, 2, 3]);
+        expect(isSubsetOf(s1, s2)).toBe(true);
+    });
+});
+
+// Helper functions are adapted from:
+// https://github.com/zloirock/core-js/blob/227a758ef96fa585a66cc1e89741e7d0bb696f48/tests/helpers/helpers.js
+
+function createSetLike<T>(elements: T[]): ReadonlySetLike<T> {
+    return {
+        size: elements.length,
+        has(value: T): boolean {
+            return includes(elements, value);
+        },
+        keys(): Iterator<T> {
+            return createIterator(elements);
+        },
+    };
+}
+
+function includes<T>(target: T[], wanted: T) {
+    for (const element of target) if (wanted === element) return true;
```

Comment:
nit: Does our prettier config consider `{}` omission and same-line body legal? Oh wow, I should take a look there :>